tests/common/Language/Java/InlineSpec.hs

```diff
@@ -130,24 +131,21 @@ spec = do
           [java| $st == Thread.State.NEW |] `shouldReturn` True
 
       it "Supports modified utf-8 encoding from Haskell to Java" $
-          withLocalRef (reflect ("a\NULb" :: T.Text)) $ \jString ->
+          withLocalRef (reflect ("a\NULb" :: Text.Text)) $ \jString ->
             [java| "a\0b".equals($jString) |] `shouldReturn` True
 
       it "Supports modified utf-8 encoding from Java to Haskell" $ do
-          withLocalRef [java| "a\0b" |] reify `shouldReturn` ("a\NULb" :: T.Text)
+          withLocalRef [java| "a\0b" |] reify `shouldReturn` ("a\NULb" :: Text.Text)
 
       prop "Processes Unicode code points as the JVM does" $ do
         \u -> ioProperty $ do
           let chars :: [Char] = fromUnicode u
-          let text = T.pack chars
-          let codePoints :: [Int32] = map (fromIntegral . ord) chars
+          let text = Text.pack chars
+          codePoints :: Vector.IOVector Int32 <- Vector.new $ length chars
+          mapM_
+            (\(i, c) -> Vector.unsafeWrite codePoints i $ fromIntegral $ ord c)
+            (zip [0..] chars)
```

Comment:
```Haskell
Data.Vector.thaw $ Data.Vector.fromList $ map ord $ fromUnicode u
```
?

Reply:
I didn't know about freeze/thaw. It's way more effective indeed